File: 查找.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
public static void main(String[] args) {
    Scanner sc=new Scanner(System.in);
    int m,n = 0;
    while(sc.hasNext()){
    	n=sc.nextInt();
    		while(n<1 || n>100)
    		{
    			n=sc.nextInt();
    		}
    		
    		int [] a=new int[n];
    		for(int i=0;i<n;i++)
    		{
    			a[i]=sc.nextInt();
    		}
    		m=sc.nextInt();
    		while(m<1 || m>100)
    		{
    			m=sc.nextInt();
    		}
    		int [] b=new int[m];
    		for(int i=0;i<m;i++)
    		{
    			b[i]=sc.nextInt();
    		}
    		
    		for(int i=0;i<m;i++)
    		{
    			if(is(a,b[i]))
    			{
    				System.out.println("YES");
    			}
    			else
    				
    			{
    				System.out.println("NO");
    			}
    		}
    		
    	}
    }
public static boolean is(int[] a,int b)
{
		for(int j=0;j<a.length;j++)
		{
			if(b==a[j])
			{
				return true;
			}
		}
	
	return false;
}
}
    		

File: 查找.cpp
#include <iostream>
using namespace std;
int main()
{
	l:
	int m,n;
	while(cin>>n)
	{
		while(n<1 || n>100)
		{
			cin>>n;
		}
		
		int a[n];
		for(int i=0;i<n;i++)
		{
			cin>>a[i];
		}
		cin>>m;
		while(m<1 || m>100)
		{
			cin>>m;
		}
		int b[n];
		for(int i=0;i<m;i++)
		{
			cin>>b[i];
		}
		for(int i=0;i<m;i++)
		{
			for(int j=0;j<n;j++)
			{
				if(b[i]==a[j])
				{
					cout<<"YES"<<endl;
					break;
				}
				else
				{
					if(b[i]!=a[j] && j==n-1)
					{
						cout<<"NO"<<endl;	
						break;
					}
				}
			}
			
		}
	}
	return 0;
}

